File: hotel-management-client/src/app/shared/wrappers/booking.wrapper.ts
import {Reservation} from '../models/reservation.model';
import {Payment} from '../models/payment.model';
import {Room} from '../models/room.model';
import {Service} from '../models/service.model';

export class Booking {
  public reservation: Reservation;
  public payment: Payment;
  public room: Room;
  public services: Service[];

  constructor() {
    this.services = [];
    this.reservation = new Reservation();
    this.payment = new Payment();
    this.room = new Room('', 0, 0, '');
  }
}

File: HotelManagementAPI/HotelManagementAPI/Entities/Wrappers/BookingWrapper.cs
using HotelManagementAPI.Entities.DTOs;

namespace HotelManagementAPI.Entities.Wrappers;

public class BookingWrapper
{
    public required ReservationDto Reservation { get; set; }
    public required PaymentDto Payment { get; set; }
    public required RoomDto Room { get; set; }
    public IEnumerable<ServiceDto>? Services { get; set; }
    
}